functions/__tests__/util.test.ts

```diff
@@ -0,0 +1,217 @@
+    )
+
+    const changeType: ChangeType = getChangeType(documentChange)
+
+    expect(changeType === ChangeType.CREATE).toBeTruthy()
+  })
+
+  test('return a update change type', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bars' },
+      'docs/1'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const changeType: ChangeType = getChangeType(documentChange)
+
+    expect(changeType === ChangeType.UPDATE).toBeTruthy()
+  })
+
+  test('return a delete change type', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/1'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const changeType: ChangeType = getChangeType(documentChange)
+
+    expect(changeType === ChangeType.DELETE).toBeTruthy()
+  })
+})
+
+describe('getDocumentId', () => {
+  const firebaseMock = firebaseFunctionsTestInit()
+
+  test('return id after create document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '2').toBeTruthy()
+  })
+
+  test('return id after update document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bars' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '2').toBeTruthy()
+  })
+
+  test('return id after delete document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '1').toBeTruthy()
+  })
+})
+
+describe('getSearchableFields', () => {
+  let util
+  let config
+  let restoreEnv
+  let mockGetSearchableFields
+
+  beforeEach(() => {
+    jest.resetModules()
+    restoreEnv = mockedEnv(environment)
+    config = global.config
```

Comment:
Does this need to be done on every test?

Reply:
Nop I can change it

Reply:
Actually, I just verify that I do this way in every test, I think it's better to have a fresh config, env, and modules for every test, no?

Reply:
Is `config` or `global.config` modified during the tests?

Reply:
Neither of them. Do you think it's better to put it outside?